src/Featurizers/AnalyticalRollingWindowFeaturizer.h

```diff
@@ -0,0 +1,337 @@
+// ----------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+// ----------------------------------------------------------------------
+#pragma once
+
+#include "../Archive.h"
+#include "../Featurizer.h"
+#include "../Traits.h"
+
+#include "Base/RollingWindowTransformerBase.h"
+#include "Calculators/MeanCalculator.h"
+#include "Components/InferenceOnlyFeaturizerImpl.h"
+#include "Components/FilterDecoratorFeaturizer.h"
+#include "Components/GrainFeaturizerImpl.h"
+#include "Components/PipelineExecutionEstimatorImpl.h"
+#include "Components/WindowFeaturizerBase.h"
+
+#include <vector>
+
+namespace Microsoft {
+namespace Featurizer {
+namespace Featurizers {
+
+static constexpr char const * const         AnalyticalRollingWindowEstimatorName("AnalyticalRollingWindowEstimator");
+static constexpr char const * const         GrainedAnalyticalRollingWindowEstimatorName("GrainedAnalyticalRollingWindowEstimator");
+
+/////////////////////////////////////////////////////////////////////////
+///  \fn            AnalyticalRollingWindowCalculation
+///  \brief         Which type of calculation will be performed on the rolling window.
+///                 Currently only one of calculation is supported. This will be expanded upon in the future,
+///                 so we are peparing by making this an enum from the get go.
+///
+enum class AnalyticalRollingWindowCalculation : std::uint8_t {
+    Mean = 1
+};
+
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         AnalyticalRollingWindowTransformer
+///  \brief         RollingWindow class that is used for all analytical window computations.
+///
+template <
+    typename InputT,
+    size_t MaxNumTrainingItemsV=std::numeric_limits<size_t>::max()
+>
+class AnalyticalRollingWindowTransformer:
+    public Base::RollingWindowTransformerBase<InputT, double, MaxNumTrainingItemsV> {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using BaseType = Base::RollingWindowTransformerBase<InputT, double, MaxNumTrainingItemsV>;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    AnalyticalRollingWindowTransformer(std::uint32_t maxWindowSize, AnalyticalRollingWindowCalculation windowCalculation, std::uint32_t horizon, std::uint32_t minWindowSize = 1) ;

```

Comment:
>AnalyticalRollingWindowTransformer [](start = 4, length = 34)

Same comments here as in `AnalyticalRollingWindowEstimator`'s constructor.

Reply:
The checking for the transformer happens in the base class. Check has been added.